### DISCUSSION TITLE
Setting cookies

### DISCUSSION THREAD

**Discussion by KolmschateS** [vercel/next.js#49971] [App Router]
I am trying to set a get and set a cookie at the same time to monitor the amount of times a user has visited a page, has to be done for a school assingment. It has to be stored in a cookie.

Next.js 13.4.2 gives me an error that I cannot use Cookies.set(): Error: ReadonlyRequestCookies cannot be modified
Which should be fixed in 13.4.2 according to this [issue](https://github.com/vercel/next.js/issues/49259). But after updating to 13.4.2 I still get the same error.

Updating to canary gives me another error:
Error: Cookies can only be modified in a Server Action or Route Handler.

Which is what I'm doing, if I'm correct.

I'm new to Next, does anybody know a fix or what I'm doing wrong?

`Vists.tsx`, a Component file
``` ts
import { cookies } from 'next/headers';

async function updateCookie()
  {
      "use server"
      // get 
      const cookieStore = cookies()

      // get visits
      const visits = Number(cookieStore.get('visits')) || 0

      // @ts-expect-error
      cookieStore.set('visits', visits + 1)

      return Promise.resolve(visits)
  }

export default function Visits()
{
    return (
        <div className='flex flex-col items-center bg-white mt-6 p-6 text-black'>
            {updateCookie()}
        </div>
        )
}
```

**Comment by moolcoov** [comment]
i have the same problem

**Reply by moolcoov** [reply]
i think it should be an issue

**Comment by suhascv** [comment]
did you try ` cookies().set('visit', visit+1)` instead of `cookieStore.set('visit', visit+1)`

https://nextjs.org/docs/app/api-reference/functions/cookies#cookiessetname-value-options
cookies() is a [Dynamic Function](https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering#using-dynamic-functions) whose returned values cannot be known ahead of time.

**Reply by KolmschateS** [reply]
Gives me this error
Error: ReadonlyRequestCookies cannot be modified

**Comment by KolmschateS** [comment]
I opened an [issue](https://github.com/vercel/next.js/issues/50009) cause I can't figure it out and to me it seems like an issue with next itself, we will see
